core/src/main/java/org/commonjava/maven/ext/manip/impl/PluginManipulator.java

```diff
@@ -63,6 +65,11 @@
     private Precedence configPrecedence = Precedence.REMOTE;
 
     /**
+     * Used to store mappings of old property to new version.
+     */
+    protected final HashMap<String, String> versionPropertyUpdateMap = new HashMap<String, String>();
```

Comment:
For reference - DependencyManipulator also has two local hashmaps it uses to track changes as it progresses through.